File: src/main/java/com/github/zyypj/tadeuBooter/api/minecraft/lang/integer/utils/NumberFormat.java
package com.github.zyypj.tadeuBooter.api.minecraft.lang.integer.utils;

import java.text.DecimalFormat;
import java.util.Arrays;
import java.util.List;

public class NumberFormat {
    private static final List<String> suffixes = Arrays.asList(
            "", "K", "M", "B", "T", "Q", "QQ", "S", "SS",
            "OC", "N", "D", "UN", "DD", "TR", "QT", "QN",
            "SD", "SPD", "OD", "ND"
    );

    /**
     * Formata um número com o sufixo apropriado da lista de sufixos.
     *
     * @param number O número a ser formatado.
     * @return Uma string representando o número formatado com sufixo.
     */
    public static String format(double number) {
        int suffixIndex = 0;

        while (number >= 1_000 && suffixIndex < suffixes.size() - 1) {
            number /= 1_000;
            suffixIndex++;
        }

        return formatNumber(number) + suffixes.get(suffixIndex);
    }

    /**
     * Formata um número com até uma casa decimal.
     *
     * @param number O número a ser formatado.
     * @return Uma string do número formatado.
     */
    private static String formatNumber(double number) {
        DecimalFormat decimalFormat = new DecimalFormat("0.#");
        return decimalFormat.format(number);
    }
}

File: src/main/java/com/github/zyypj/tadeuBooter/api/minecraft/lang/integer/utils/NumberFormat.kt
package com.github.zyypj.tadeuBooter.api.minecraft.lang.integer.utils

import java.text.DecimalFormat
import kotlin.jvm.JvmStatic

/**
 * Formata números com sufixos (K, M, B, ...).
 * Métodos anotados com @JvmStatic estão disponíveis como métodos estáticos em Java:
 * NumberFormat.format(12345.0)
 */
object NumberFormat {

    private val suffixes = listOf(
        "", "K", "M", "B", "T", "Q", "QQ", "S", "SS",
        "OC", "N", "D", "UN", "DD", "TR", "QT", "QN",
        "SD", "SPD", "OD", "ND"
    )

    /**
     * Formata um número com o sufixo apropriado.
     * Exemplo: 1500.0 -> "1.5K"
     */
    @JvmStatic
    fun format(number: Double): String {
        var n = number
        var index = 0
        while (n >= 1_000 && index < suffixes.lastIndex) {
            n /= 1_000
            index++
        }
        return formatNumber(n) + suffixes[index]
    }

    /**
     * Formata número com até uma casa decimal.
     */
    @JvmStatic
    private fun formatNumber(number: Double): String {
        val df = DecimalFormat("0.#")
        return df.format(number)
    }
}